### DISCUSSION TITLE
Things that are coming, and things that could come, but not unless people squawk for them

### DISCUSSION THREAD

**Discussion by SpenceKonde** [SpenceKonde/DxCore#108] [Future development discussion]
Very near future:
There will be a new version coming ASAP. This will have the new pymcuprog based tools, plus:
* keywords will contain register names so they'll get highlighted 
* New TWI baud calculation. which should actually be accurate. 
* Set of #defines relating to which pins are used by peripherals under various circumstances. 
* fixes and tweaks relating to the ADC stuff that went in recently. 
* The pinConfig function, which allows every parameter of a pin to be set to either of the options, left unchanged, and in most cases toggled. 
* A few updates to some libraries I think are also involved. But yeah tying up loose ends here. 
* Look into the curiosity nano serial port thing.

AFTER THAT RELEASE
there will be a pause in development over here while I turn my attention to finishing ATTinyCore 2.0.0. After that, there will likely be no ATTinyCore 2.1.0 - just patch-level bugfixes. But I owe it to those users to finish it, and to myself, as I've spent a ton of time already on getting as close as I am now to this. The core dev is nearly done, then there's some bootloader stuff, 
I'll only be back here in the course of trying to make sure we have CI testing when things start up. AI have a ton of users over there.. 

After that, the next release will address bugs discovered since then, any particular new feature requests, as well as 
THE SLEEP LIBRARY
and the
DD series support assuming they are looking like they're actually getting close to availability. See below for some notes on that. 

DD-series
DD-series parts seem to be taking their sweet time. Probably because Microchip has their hands and fabs full (failing to) keep up with demand for their existing product lines. For example, I could use some more ATtiny1616's in the QFN package. All the major distributors are out of stock, and backorderes from, digikey and mouser are expected to ship in... oh.... september maybe? lol

Currently things are mostly ready (though not checked in, in many cases) for the DD-series.... with the exception of the USART support. The core was implemented with the idea that there would be only be two pinsets. So that functionality needs to be totally rewritten. Everything else is basically done for the DD-series. Depending on how it goes, there may bne a stopgap release with pml;y two mappings per USART (there is also a larger issue with the USART implementation, more relevant to megaTinyCore, namely that it's grotesquely bloated and people are complaining about the flash size. I mean, I can't really blame them either when they are unhappy coming here from a 1k flash tiny13, where a softwarwe serial fit easily, and then step into megaTinyCore, and our HARDWARE serial implementation, to initialize a port and print a single character, results in a 1700 byte sketch. And looking though the memory map and .lst. yes, it is AWFUL. It is dreadful. An abomination. Serious work is needed on that front too. (jeez, and I ALREADY did MAJOR WORK on it). The official Arduino serial implementation was just unspeakably bad. 

There are also some other things that could exist, but I don't think anyone cares: 
* #41 - do people want weird millis timekeeping options tht trade multiple timeers for fully functional non-interrupt millis? 
* Demo of rerouting serial inputs and outputs with eventr and logic systems
* Demos of doing crazy stuff with logic and event systems. 
* Do people want/need demos of use of any peripherals not yet covered by library? Input capture? Something else?

**Comment by gordonthree** [comment]
I'll add my squawk rather lately, for the last question in your post, "Do people want/need demos of use of any peripherals not yet covered by library? Input capture? Something else?"

I'd love to see an example for TWI_MANDS_SINGLE ... for the 28 pin chip especially, but any chip will do.
